### PR TITLE
Single ship view page

### LINKED ISSUE #8: View ship data

### PR DESCRIPTION
**Pull request by ljweston**
Closes #8 

Displays ship data and has navigation button to go too and leave the page.

### CODE REVIEW

File: manageShips.php [hunk 1]
```diff
@@ -35,7 +35,8 @@
                 <tbody>
                     <?php foreach ($ships as $ship): ?>
                         <tr>
-                            <td><a href=""><?php echo $ship->getName(); ?></a></td>
+                            <td><a href="viewShip.php?id=<?php echo $ship->getId();?>">
```
Comment (mlellis) [inline manageShips.php:38]: I think this should have the `/` at the beginning of the route

File: viewShip.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+use Service\Container;
+
+
+
+$id = $_GET['id'];
+
+// Need shiploader obj for ship
+$container = new Container($configuration);
+$shipLoader = $container->getShipLoader();
+$ship = $shipLoader->findOneById($id); // values may be getting changed here
```
Comment (mlellis) [inline viewShip.php:14]: There should be some error checking around this:

1. Make sure the 'id' parameter is in the GET request
2. Show a message a ship is not found (in invalid id is passed in)

File: viewShip.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+require 'layout/header.php';
+
+?>
+
+    <div class="container">
+            <div class="page-header">
```
Comment (mlellis) [inline viewShip.php:21]: I think a breadcrumb would be perfect for this. 

Something like: `Manage Ships > [shipName] Profile` and 'Manage Ships' is a link back to the manage page.

Then you can add onto this for the edit in the future: `Manage Ships > [shipName] Profile > Edit`

File: viewShip.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                <h1>Viewing the <?php echo $ship->getName()?></h1>
+            </div>
+            <a href="/manageShips.php" class="btn btn-md btn-primary pull-right">Manage Ships</a>
+            
+            <table class="table table-striped">
+                <tbody>
+                    <tr class="d-flex">
```
Comment (mlellis) [inline viewShip.php:28]: Are you using the `d-flex` and `col-1` here to make the first column smaller? If so, I think an easier approach would be to add something like `style="width: 100px;"` to the first `th` element. The others will follow the same width.

File: viewShip.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                        <th class="col-1">Jedi Power:</th>
+                        <td><?php echo $ship->getJediFactor()?></td>
+                    </tr>
+                </tbody>
+            </table>
+            <div>
+                <a href="" class="btn btn-md btn-success">Edit</a>
+                <a href=""class="btn btn-md btn-danger">Delete</a>
+            </div>
+    </div>
```
Comment (mlellis) [inline viewShip.php:50]: I think somewhere on this page (in the table or elsewhere) should show if the ship needs repair.

When we get to the add page, I think you can store the health status of the ship. Then show that here too.

**Review by mlellis — CHANGES_REQUESTED** [review]

File: viewShip.php (hunk 3 above)
Comment (ljweston) [reply viewShip.php:45]: Put in a dummy breadcrumb trail. Working on implementation for a fully fledged one.

File: manageShips.php [hunk 6]
```diff
@@ -35,7 +39,8 @@
                 <tbody>
                     <?php foreach ($ships as $ship): ?>
                         <tr>
-                            <td><a href=""><?php echo $ship->getName(); ?></a></td>
+                            <td><a href="/viewShip.php?id=<?php echo $ship->getId();?>">
```
Comment (mlellis) [inline manageShips.php:42]: I think split these out into separate lines:

```
<td>
   <a ...>...</a>
</td>
```

File: manageShips.php [hunk 7]
```diff
@@ -16,6 +16,10 @@
     <!-- Error handling -->
 
         <div class="container">
+            <ul class="breadcrumb">
+                <li><a href="/index.php">Home</a></li>
+                <li><a>ManageShips</a></li>
```
Comment (mlellis) [inline manageShips.php:21]: Remove the empty `<a>` tag here

**Review by mlellis — CHANGES_REQUESTED** [review]

**Pull request merged**